client.go

```diff
@@ -2418,8 +2418,7 @@ func (c *Client) ImageFromContainer(cname string, public bool, aliases []string,
 	if c.Remote.Public {
 		return "", fmt.Errorf("This function isn't supported by public remotes.")
 	}
-
-	source := shared.Jmap{"type": "container", "name": cname}
+	source := shared.Jmap{"type": "container", "name": cname, "compression": properties["compression"]}
```

Comment:
The compression algorithm should be an extra argument to ImageFromContainer, not abusing the properties map which is used for user defined metadata on the image.